### PR TITLE
Stories/137 mturkrecruiter

### PR DESCRIPTION
**Pull request by jessesnyder**
First stage of an MTurkRecruiter implemented directly on boto.mturk, without psiturk.

## Description
At this point the MTurkRecruiter only implements the `open_recruitment()` method, but provides a context for completing a full implementation of the Recruiter interface.

## Motivation and Context
Another step along the road to being free of psiturk.

## How Has This Been Tested?
There are both isolation tests, which mock out boto.mturk code in interactions, and full system tests
which actually call MTurk and create HITs. The system tests check for three environment variables, and are skipped if they're not present:

- aws_access_key_id
- aws_secret_access_key
- is_aws_mturk_requestor

The key id and secret must be for an AWS account connected to an MTurk "Requestor" account. The is_aws_mturk_requestor variable is checked since there may be other tests that require an AWS login
but don't required that the account be linked to a requestor.

Ideally the system tests would be run in the CI test suite, but getting this set up will require some time and thought.

The MTurkRecruiter is **not** yet hooked up to the rest of the system. To do this, you would replace
`PsiTurkRecruiter` with `MTurkRecruiter.from_current_config` in the `recruiter` property of `experiment.Experiment`.

There are currently some unrelated tests failing because something is trying to access the "turkdemo" db table, which exists in the psiturk db but not the dallinger db.

### CODE REVIEW

File: dallinger/mturk.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        if not self.aws_access_key_id or not self.aws_secret_access_key:
+            raise MTurkServiceException('AWS access key and secret not set.')
+        login_params = {
+            'aws_access_key_id': self.aws_access_key_id,
+            'aws_secret_access_key': self.aws_secret_access_key,
+            'host': self.host
+        }
+        if self._connection is None:
+            self._connection = MTurkConnection(**login_params)
+        return self._connection
```
Comment (davisagli) [inline dallinger/mturk.py]: @jessesnyder This caching would be cleaner if we borrow Pyramid's `reify` decorator: http://docs.pylonsproject.org/projects/pyramid/en/latest/_modules/pyramid/decorator.html#reify

Comment (jessesnyder) [inline dallinger/mturk.py]: That's very nice; I'll add reify to utils.py and use it.

File: tests/test_clock.py [hunk 2]
```diff
@@ -26,9 +26,8 @@ def test_scheduler_has_job(self):
     def test_clock_expects_config_to_be_ready(self):
         assert not get_config().ready
         jobs = self.clock.scheduler.get_jobs()
-        with raises(RuntimeError) as excinfo:
+        with raises(RuntimeError, message='Config loading not finished'):
```
Comment (MatthewWilkes) [inline tests/test_clock.py]: This is cool, thanks!

Comment (jessesnyder) [inline tests/test_clock.py]: Yeah, totally cool. Thanks for making the case for pytest. I'd been of course hearing a lot of buzz but hadn't actually used it before now.

Comment (MatthewWilkes) [inline tests/test_clock.py]: I'd been the same. I thought it was a bit icky when I first heard the idea, but then when I worked on a project that used it the usability really outweighed any downsides.

Comment (jessesnyder) [inline tests/test_clock.py]: @MatthewWilkes Except... it doesn't work the way I thought/hoped. It's just a way to set an alternative exception message of course. <face palm>

**Pull request closed without merging**